File: YunLog/src/main/java/com/example/yunlog/java/YunConsolePrinter.java
package com.example.yunlog.java;

import android.util.Log;

import androidx.annotation.NonNull;

import static com.example.yunlog.java.YunLogConfig.LINE_MAX_LENGTH;

class YunConsolePrinter implements YunLogPrinter {
    @Override
    public void print(@NonNull YunLogConfig config, int level, String tag, @NonNull String printString) {
        int len = printString.length();
        int depth = len / LINE_MAX_LENGTH;

        if (depth == 0) {
            Log.println(level, tag, printString);
        } else {
            int index = 0;
            for (int i = 0; i < depth; i++) {
                Log.println(level, tag, printString.substring(index, index + LINE_MAX_LENGTH));
                index += LINE_MAX_LENGTH;
            }
            if (index != len) {
                Log.println(level, tag, printString.substring(index, len));
            }
        }

    }
}

File: YunLog/src/main/java/com/example/yunlog/kotlin/YunConsolePrinter.kt
package com.example.yunlog.kotlin

import android.util.Log
import com.example.yunlog.java.YunLogConfig.LINE_MAX_LENGTH

open class YunConsolePrinter : YunLogPrinter {
    override fun print(config: YunLogConfig, type: Int, tag: String?, printString: String) {
        val len = printString.length
        val depth = len / LINE_MAX_LENGTH

        when (depth) {
            0 -> {
                Log.println(type, tag, printString)
            }
            else -> {
                var index = 0
                for (subLength in 0 until depth-1) {
                    Log.println(type, tag, printString.substring(index, index + LINE_MAX_LENGTH))
                    index += LINE_MAX_LENGTH
                }

                if (index != len) {
                    Log.println(type, tag, printString.substring(index, len))
                }
            }
        }

    }
}